components/playlists/browser/playlists_controller.h

```diff
@@ -0,0 +1,177 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PLAYLISTS_BROWSER_PLAYLISTS_CONTROLLER_H_
+#define BRAVE_COMPONENTS_PLAYLISTS_BROWSER_PLAYLISTS_CONTROLLER_H_
+
+#include <list>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/callback_forward.h"
+#include "base/containers/queue.h"
+#include "base/macros.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
+#include "base/values.h"
+#include "brave/components/playlists/browser/playlists_media_file_controller.h"
+#include "brave/components/playlists/browser/playlists_types.h"
+
+namespace base {
+class FilePath;
+class SequencedTaskRunner;
+}  // namespace base
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+namespace brave_playlists {
+
+class PlaylistsControllerObserver;
+class PlaylistsDBController;
+class PlaylistsPlayer;
+
+class PlaylistsController : PlaylistsMediaFileController::Client {
+ public:
+  explicit PlaylistsController(content::BrowserContext* context);
+  ~PlaylistsController() override;
+
+  bool initialized() const { return initialized_; }
+  bool initialization_in_progress() const {
+    return initialization_in_progress_;
+  }
+
+  bool Init(const base::FilePath& base_dir,
+            scoped_refptr<base::SequencedTaskRunner> task_runner);
+
+  void set_player(PlaylistsPlayer* player) { player_ = player; }
```

Comment:
why does a controller need to own a player? We have service -> controller -> player? I don't understand why we need all these layers, it just seems confusing to me.